### PR TITLE
DA-574 refactoring to pull import organizations into dao's

### PR DESCRIPTION
**Pull request by Seeker1911**

### CODE REVIEW

File: rest-api/tools/csv_importer.py [hunk 1]
```diff
@@ -113,14 +113,14 @@ def _update_entity(self, entity, existing_entity, session, dry_run):
...
-        self._do_update(entity, existing_entity, session)
+        self.dao._do_update(session, entity, existing_entity)
       return True, False
 
-  def _do_update(self, entity, existing_entity, session):
-    for k, v in entity.asdict().iteritems():
-      if k != self.external_id_field and k != self.id_field:
-        setattr(existing_entity, k, v)
-    self.dao.update_with_session(session, existing_entity)
+#  def _do_update(self, session, entity, existing_entity):
```
Comment (danqrodney) [inline rest-api/tools/csv_importer.py]: You need this. csv_importer shouldn't have to change.

File: rest-api/test/unit_test/dao_test/organization_dao_test.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    created_organization = self.organization_dao.insert(organization)
+    new_organization = self.organization_dao.get(created_organization.organizationId)
+    organization.organizationId = created_organization.organizationId
+    self.assertEquals(organization.asdict(), new_organization.asdict())
+
+  def test_participant_pairing_updates_onchange(self):
+    insert_org = self.organization_dao.insert(
+                  Organization(externalId='tardis', displayName='bluebox', hpoId=PITT_HPO_ID))
+
+    self.participant_dao.insert(Participant(participantId=1, biobankId=2))
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/unit_test/dao_test/organization_dao_test.py]: done

File: rest-api/test/unit_test/dao_test/organization_dao_test.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    participant = self.participant_dao.get(1)
+    participant.organizationId = insert_org.organizationId
+    self.participant_dao.update(participant)
+
+    self.assertEquals(participant.hpoId, insert_org.hpoId)
+    insert_org.hpoId = AZ_HPO_ID
+    self.organization_dao.update(insert_org)
+    new_org = self.organization_dao.get_by_external_id('tardis')
+    participant = self.participant_dao.get(1)
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/unit_test/dao_test/organization_dao_test.py]: done

File: rest-api/test/unit_test/dao_test/organization_dao_test.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    self.assertNotEquals(participant.hpoId, PITT_HPO_ID)
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/unit_test/dao_test/organization_dao_test.py]: done

File: rest-api/dao/organization_dao.py [hunk 5]
```diff
@@ -24,6 +25,48 @@ def _validate_update(self, session, obj, existing_obj):
...
     pass
 
+  def _do_update(self, session, obj, existing_obj):
+    update_participants = False
+    if obj.hpoId != existing_obj.hpoId:
+      update_participants = True
+      new_hpo_id = obj.hpoId
+    super(OrganizationDao, self)._do_update(session, obj, existing_obj)
+    if update_participants:
+      from participant_enums import make_primary_provider_link_for_id
```
[the remark this answers is not in the document]
Comment (danqrodney) [reply rest-api/dao/organization_dao.py]: Ping.

File: rest-api/data/sites.csv [hunk 6]
```diff
@@ -31,7 +31,7 @@ CAL_PMC_UCI,hpo-site-uofcirvinesantaana,"University of California, Irvine  Dept
...
 CAL_PMC_UCSF,hpo-site-uofcsanfrancisco,University of California San Fransico Mt. Zion Campus,HPO U of C San Francisco,7036539,4/16/2018,,Active,Active,Active,Active,Inactive,Inactive,Active,Inactive,Inactive,Active,Inactive,Inactive,,,Mt. Zion Hoispital,1600 Dovosaderp St.,A645,San Francisco,CA ,94115,,"escott@psg.ucsf.edu, julissa.cabrera@ucsf.edu",
-CAL_PMC_CEDARS_SINAI,hpo-site-cedarssinaimedicalcenter,Cedar-Sinai Medical Center,HPO Cedars Sinai Medical Center,7036572,4/16/18,,Active,Active,Active,Active,Inactive,Inactive,Active,Inactive,Inactive,Active,Inactive,Inactive,,,,8700 Beverly Blvd,,Los Angeles,CA ,90048,,"Christina.lombardi@cshs.org, melanie.huynh@cshs.org ",
+CAL_PMC_CEDARS_SINAI,hpo-site-cedarssinaimedicalcenter,Cedars-Sinai Medical Center,HPO Cedars Sinai Medical Center,7036572,4/16/18,,Active,Active,Active,Active,Inactive,Inactive,Active,Inactive,Inactive,Active,Inactive,Inactive,,,,8700 Beverly Blvd,,Los Angeles,CA ,90048,,"Christina.lombardi@cshs.org, melanie.huynh@cshs.org ",
```
[the remark this answers is not in the document]
Comment (danqrodney) [reply rest-api/data/sites.csv]: OK. Normally we'd have a separate PR for updating sites data. (Not a big deal, but is easier to track that way.)

There's nothing in this PR that requires a change here is there?

File: rest-api/test/unit_test/dao_test/site_dao_test.py [hunk 7]
```diff
@@ -36,3 +43,27 @@ def test_update(self):
...
+    update_site_parent = Site(siteId=created_site.siteId, siteName='site2',
+                    googleGroup='site2@googlegroups.com',
+                    mayolinkClientNumber=123456, hpoId=AZ_HPO_ID, organizationId=AZ_ORG_ID)
+    self.site_dao.update(update_site_parent)
+    updated_p = self.participant_dao.get(fetch_p.participantId)
+    p_summary = self.ps_dao.insert(self.participant_summary(updated_p))
+    ps = self.ps_dao.get(p_summary.participantId)
+    ph = self.ps_history.get([updated_p.participantId, 1])
+
+    self.assertEquals(update_site_parent.hpoId, updated_p.hpoId)
```
[the remark this answers is not in the document]
Comment (danqrodney) [reply rest-api/test/unit_test/dao_test/site_dao_test.py]: Ping.

File: rest-api/test/unit_test/dao_test/organization_dao_test.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (danqrodney) [reply rest-api/test/unit_test/dao_test/organization_dao_test.py]: Please check last_modified on the summary. (Note you may need to use a FakeClock when doing the update above to have the right assert.)

File: rest-api/test/unit_test/dao_test/site_dao_test.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/unit_test/dao_test/site_dao_test.py]: done

File: rest-api/test/unit_test/dao_test/organization_dao_test.py [hunk 8]
```diff
@@ -23,23 +25,25 @@ def test_insert(self):
...
   def test_participant_pairing_updates_onchange(self):
+    TIME = datetime.datetime(2018, 1, 1)
     insert_org = self.organization_dao.insert(
                   Organization(externalId='tardis', displayName='bluebox', hpoId=PITT_HPO_ID))
 
-    self.participant_dao.insert(Participant(participantId=1, biobankId=2))
-    participant = self.participant_dao.get(1)
-    participant.organizationId = insert_org.organizationId
-    self.participant_dao.update(participant)
+    with FakeClock(TIME):
```
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/test/unit_test/dao_test/organization_dao_test.py]: done

File: rest-api/data/sites.csv (hunk 6 above)
[the remark this answers is not in the document]
Comment (Seeker1911) [reply rest-api/data/sites.csv]: no, and this same change is in the new import by CH, I just ran an export_orgs in testing and this got pulled in.

**Pull request merged**